Gem/Code/Source/GameLift/GameLiftClientComponent.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzFramework/Matchmaking/MatchmakingNotifications.h>
+
+#include <Request/AWSGameLiftRequestBus.h>
+#include <Request/AWSGameLiftSessionRequestBus.h>
+#include <Request/AWSGameLiftMatchmakingRequestBus.h>
+
+
+namespace MultiplayerSample
+{
+    class GameLiftClientSystemComponent
+        : public AZ::Component
+        , public AzFramework::SessionAsyncRequestNotificationBus::Handler
+        , public AzFramework::MatchmakingNotificationBus::Handler
+        , public AzFramework::MatchmakingAsyncRequestNotificationBus::Handler
+    {
+    public:
+        AZ_COMPONENT(GameLiftClientSystemComponent, "{F83C5B49-3E33-4BB9-B536-0232D310C298}");
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
+    protected:
+        ////////////////////////////////////////////////////////////////////////
+        // AZ::Component interface implementation
+        void Init() override;
+        void Activate() override;
+        void Deactivate() override;
+
+        void HostSession(const AZ::ConsoleCommandContainer& consoleFunctionParameters);

```

Comment:
nit: Consider adding brief comments to this method esp as you are providing functionality and registered console funcs